scene/animation/animation_player.cpp

```diff
@@ -1205,9 +1205,13 @@ void AnimationPlayer::play(const StringName &p_name, float p_custom_blend, float
 	_stop_playing_caches();
 
 	c.current.from = &animation_set[name];
-	c.current.pos = p_from_end ? c.current.from->animation->get_length() : 0;
+
+	if (c.assigned != name) { // reset
```

Comment:
Thought some more about this, and there would still be one problematic case with this change I proposed. It makes it quite hard to reset a backward-playing animation, since `stop(true)` will reset it to 0, and `play_backwards()` or `play_backwards("same_animation")` would now resume from the `c.current.pos`, which is 0, so the animation would end right away.

Not sure if there's any good way to handle this properly.

Reply:
One option could be to add a `bool p_reset = false` parameter (both to `play` and `play_backwards`), and use it here:
```
if (c.assigned != name || p_reset) { // reset
	c.current.pos = p_from_end ? c.current.from->animation->get_length() : 0;
}
```

The other option is to consider that `p_from_end` always means "reset to end", so it could be:
```
if (p_from_end) {
	c.current.pos = c.current.from->animation->get_length();
} else if (c.assigned != name) { // reset
	c.current.pos = 0;
}
```
That would make it harder to resume a backwards-playing animation though, you'd have to do `play("animation_name", custom_blend, -1, false)`.

Reply:
What about this:

``` C++
if (c.assigned != name) { // reset
	c.current.pos = p_from_end ? c.current.from->animation->get_length() : 0;
} else if (c.current.pos == 0 && p_from_end) { // Animation reseted BUT played backwars, set position to the end
	c.current.pos = c.current.from->animation->get_length()
}
```

When calling `play_backwars()` after calling `stop(true)` the animation will be at position 0 inevitably, so we just start from the end, just like we do when calling `play_backwars("different_animation")`.

Reply:
Yeah I was thinking about something like that too, that's a good idea.

Reply:
Style wise, I'd just do the check the other way around though (so `else if (p_from_end && c.current.pos == 0)`.
And typo fixes for the suggested comment:
"// Animation reset BUT played backwards, set position to the end"

Reply:
It would be great if you could ensure that the documentation (in `doc/classes/AnimationPlayer.xml`) properly describes how things work after this fix.

Reply:
Cool, I can do this. I'm trying my best with English, will make these adjustments and then check the docs. It should be another PR or is it fine to put it here?

Reply:
It would be best if it's in the same commit, so that the logic change and corresponding documentation change are together.

Reply:
On it.

Reply:
I hope the documentation update is clear enough.